metricflow/test/table_snapshot/table_snapshots.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import annotations
+
+import datetime
+import logging
+import os
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import Tuple, Sequence, List, Union, Optional
+
+import dateutil.parser
+import pandas as pd
+import yaml
+
+from dbt_semantic_interfaces.objects.base import FrozenBaseModel
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.object_utils import hash_items, assert_values_exhausted
+from metricflow.protocols.sql_client import SqlClient
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class SqlTableSnapshotHash:
+    """A hash that can be used to compare whether a SQL table snapshot has changed."""
+
+    str_value: str
+
+    @staticmethod
+    def create_from_hashes(hashes: Sequence[SqlTableSnapshotHash]) -> SqlTableSnapshotHash:  # noqa: D
+        return SqlTableSnapshotHash(hash_items(tuple(one_hash.str_value for one_hash in hashes)))
+
+
+class SqlTableColumnType(Enum):  # noqa: D
```

Comment:
How many sql type enums do we have now? Three?

I don't know if we should accumulate more calls for the others, so maybe this is ok, but it does pop up quite often.

Reply:
What was the other one?

Reply:
There's one in inference and another that has something to do with bind parameters.

Reply:
They're not quite equivalent use cases, so need to have a think about how to consolidate those.

Reply:
Yeah, it's mostly annoying. No big deal.